src/actions/receive-profile.js

```diff
@@ -77,10 +77,23 @@ export function coalescedFunctionsUpdate(
   };
 }
 
-const requestIdleCallbackPolyfill: typeof requestIdleCallback =
-  typeof window === 'object' && window.requestIdleCallback
-    ? window.requestIdleCallback
-    : callback => setTimeout(callback, 0);
+// This polyfill looks crazy, but it's mainly to satisfy the type signature for Flow.
+// It's essentially a setTimout(callback, 0).
+let requestIdleCallbackPolyfill = (
+  callback: (deadline: {
+    didTimeout: boolean,
+    timeRemaining: () => number,
+  }) => void,
+  _opts?: { timeout: number }
+) => {
+  setTimeout(() => {
+    callback({ didTimeout: false, timeRemaining: () => 0 });
+  }, 0);
+};
+
+if (typeof window === 'object' && window.requestIdleCallback) {
+  requestIdleCallbackPolyfill = window.requestIdleCallback;
+}
```

Comment:
Given how we use it (especially we don't use the callback's parameters), why don't you simplify this to just what we need? I tried this and it works fine:
```javascript
let requestIdleCallbackPolyfill = (
  callback: () => void,
  _opts?: { timeout: number }
) => {
  setTimeout(callback, 0);
};

if (typeof window === 'object' && window.requestIdleCallback) {
  requestIdleCallbackPolyfill = window.requestIdleCallback;
}
```

I noticed we need to keep the `_opts` so that flow types the variable properly.

Maybe here is another possibility, possibly simpler, that also works:
```javascript
let requestIdleCallbackPolyfill: (
  callback: () => void,
  _opts?: { timeout: number }
) => mixed;

if (typeof window === 'object' && window.requestIdleCallback) {
  requestIdleCallbackPolyfill = window.requestIdleCallback;
} else {
  requestIdleCallbackPolyfill = callback => setTimeout(callback, 0);
}
```

Reply:
Thanks! Much simpler.